lib/build-plugin.js

```diff
@@ -33,7 +33,12 @@ function buildPlugin(Client, pluginOpts) {
     const concurrency = opts.concurrency || DEFAULT_GET_CONCURRENCY
 
     const secrets = Array.isArray(opts.secrets)
-      ? await pReduce(opts.secrets, async (acc, value) => ({ ...acc, [value]: await client.get(value) }), {})
+      ? Object.assign(
+          {},
+          ...(await pMap(opts.secrets, async (value) => ({ [value]: await client.get(value) }), {
+            concurrency
+          }))
+        )
```

Comment:
No need to change, but a possible alternative could have been
```suggestion
      ? Object.fromEntries(
           await pMap(opts.secrets, async (value) => [ [value], await client.get(value) ], { concurrency })
        )
```